datafusion/doc-gen/src/lib.rs

```diff
@@ -190,7 +177,7 @@ impl DocumentationBuilder {
         self
     }
 
-    pub fn build(self) -> Result<Documentation> {
+    pub fn build(self) -> Documentation {
```

Comment:
Perhaps it is worth documenting that this will panic if doc section, description, or syntax example is missing

Reply:
Done. I was thinking to convert the builder to support constructor `new` with mandatory arguments so we dont need having this panics, I'll do it in follow up PR

Reply:
I like that idea